### PR TITLE
start work on dsp subsystem

### PR DESCRIPTION
**Pull request by Victor-M-Giraldo**

### CODE REVIEW

File: figures/dsp/dsp-arq.mmd [hunk 1]
```diff
@@ -0,0 +1,15 @@
+sequenceDiagram
+    participant Sender
+    participant Receiver
+
+    Sender->>Receiver: Transmit Packet
+    Note over Sender, Receiver: Packet transmission
+    alt Packet received successfully
+        Receiver->>Sender: ACK
+        Note over Sender, Receiver: Acknowledgment received
+    else Packet lost or corrupted
```
Comment (dyc3) [inline figures/dsp/dsp-arq.mmd:10]: How does the receiver know if the packet is lost? They never got it, so they don't know its there.

Comment (Victor-M-Giraldo) [reply figures/dsp/dsp-arq.mmd]: I would assume if the request times out or like you said if there is an absence of the packet.

Comment (Victor-M-Giraldo) [reply figures/dsp/dsp-arq.mmd]: Ah okay I get what you mean

File: figures/dsp/dsp-arq.mmd [hunk 2]
```diff
@@ -0,0 +1,16 @@
+sequenceDiagram
+    participant Sender
+    participant Receiver
+
+    Sender->>Receiver: Transmit Packet
+    Note over Sender, Receiver: Packet transmission
```
Comment (dyc3) [inline figures/dsp/dsp-arq.mmd:6]: nit: these notes are excessive and just make the diagram harder to read. remove them

**Comment by Victor-M-Giraldo** [comment]
```mermaid
sequenceDiagram
    participant Sender
    participant Receiver

    Sender->>Receiver: Transmit Packet
    alt Packet received successfully
        Receiver->>Sender: ACK
    else Timeout
        Sender->>Sender: Retransmit Packet
    end

```

Something like this better?

**Comment by dyc3** [comment]
Yeah

**Review by dyc3 — APPROVED** [review]

**Pull request merged**